packages/storage/amplify_storage_s3_dart/lib/src/model/s3_client_config_copy.dart

```diff
@@ -0,0 +1,24 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import 'package:aws_signature_v4/aws_signature_v4.dart' as sigv4;
+import 'package:smithy_aws/smithy_aws.dart' as smithy_aws;
+
+/// Extension adds a copyWith method to [smithy_aws.S3ClientConfig] allowing
+/// to create a new instance of [smithy_aws.S3ClientConfig] with overrides.
+extension S3ClientConfigCopy on smithy_aws.S3ClientConfig {
+  /// Returns a new [smithy_aws.S3ClientConfig] with overrides.
+  smithy_aws.S3ClientConfig copyWith({
```

Comment:
Can you add this to `S3ClientConfig` instead?

Reply:
Updated 👍